src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -171,34 +189,42 @@ private void handleExit() {
      * Changes theme to light theme
      */
     @FXML
-    private void handleLightTheme() {
-        boolean hasLightTheme = primaryStage.getScene().getStylesheets().contains(lightTheme);
-        boolean hasDarkTheme = primaryStage.getScene().getStylesheets().contains(darkTheme);
-
-        if (!hasLightTheme && hasDarkTheme) {
-            primaryStage.getScene().getStylesheets().remove(darkTheme);
+    private void handleLightDarkTheme() {
+        if (isLightTheme) {
+            logger.info("Switching to dark theme...");
+            isLightTheme = false;
+            lightDarkThemeItem.setText(LIGHT_THEME_MENUITEM_TEXT);
+            primaryStage.getScene().getStylesheets().add(darkTheme);
+            primaryStage.getScene().getStylesheets().add(extensionsDark);
+            primaryStage.getScene().getStylesheets().remove(lightTheme);
+            primaryStage.getScene().getStylesheets().remove(extensionsLight);
+            helpWindow.setDarkTheme();
+        } else {
+            logger.info("Switching to light theme...");
+            isLightTheme = true;
+            lightDarkThemeItem.setText(DARK_THEME_MENUITEM_TEXT);
             primaryStage.getScene().getStylesheets().add(lightTheme);
-            primaryStage.getScene().getStylesheets().remove(extensionsDark);
             primaryStage.getScene().getStylesheets().add(extensionsLight);
+            primaryStage.getScene().getStylesheets().remove(darkTheme);
+            primaryStage.getScene().getStylesheets().remove(extensionsDark);
+            helpWindow.setLightTheme();
         }
-        helpWindow.setLightTheme();
     }
 
-    /**
-     * Changes theme to dark theme
-     */
     @FXML
-    private void handleDarkTheme() {
-        boolean hasLightTheme = primaryStage.getScene().getStylesheets().contains(lightTheme);
-        boolean hasDarkTheme = primaryStage.getScene().getStylesheets().contains(darkTheme);
-
-        if (!hasDarkTheme && hasLightTheme) {
-            primaryStage.getScene().getStylesheets().remove(lightTheme);
-            primaryStage.getScene().getStylesheets().add(darkTheme);
-            primaryStage.getScene().getStylesheets().remove(extensionsLight);
-            primaryStage.getScene().getStylesheets().add(extensionsDark);
+    private void handleCompactExpand() {
```

Comment:
May want to consider adding Javadocs for `handleCompactExpand` too.

Reply:
Yea will do.